content/blog/2019/root-transition/index.md

```diff
@@ -0,0 +1,106 @@
+As a best practice for security, we strongly recommend you to do annual root key/cert rotation.
+We will send out instructions for root key/cert rotation as a follow-up.
+
+To evaluate the lifetime remaining for your root certificate, please refer to Step 0 in the following procedure.
+
+To resolve this problem, we have a procedure that does the root transition automatically.
+Note that the Envoy instances will be hot restarted to reload the new root certificates, which may impact the long-lived connections.
+For details about the impacts and how Envoy hot restart works, please refer to
+[here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/hot_restart#arch-overview-hot-restart) and
+[here](https://blog.envoyproxy.io/envoy-hot-restart-1d16b14555b5).
+
+## Scenarios
+
+If you are not currently using the mTLS feature in Istio and will not use it in the future, you are not affected and can skip the below content.
+
+If you are not currently using the mTLS feature in Istio and may use it in the future, you are recommended to follow the procedure listed below to upgrade. 
+
+If you are currently using the mTLS feature in Istio, please follow the procedure and check whether you will be affected.
+
+## Procedure
+
+### Step 0. Check when your root certificate will expire
+
+Download this [script](TODO) on a machine that has kubectl access to the cluster.
+
+{{< text bash>}}
+$ ./root-transition.sh check
+...
+===YOU HAVE 364 DAYS BEFORE THE ROOT CERT EXPIRED!=====
+{{< /text >}}
+
+If there is less than 30 days remaining, we suggest to run following steps immediately to avoid system outage. Otherwise we will have a smoother solution released in 30 days.
+
+### Step 1. Do the root certificate transition
+
+Run the following command to do the root certificate transition.
+
+{{< text bash>}}
+$ ./root_transition.sh transition
+{{< /text >}}
+
+After the transition, the Envoy sidecars may be hot-restarted. This does not have an impact on short-lived connections. Long-lived connections will be gracefully drained, and normally applications can reconnect automatically.
+
+### Step 2. Verify the new workload certificates are generated
+
+Run the following command to verify that all the new certificates are generated. If not, wait a minute and run it again.
+
+{{< text bash>}}
+$ ./root_transition.sh verify
+{{< /text >}}
+
+### Step 3. Install Istio 1.1.8
+
+Note that this step is necessary, to make sure the control plane components and sidecar Envoys all reload the new certificates and keys.
+
+Upgrade your control plane and Envoy sidecars to 1.1.8. You can refer to this
+[doc](https://istio.io/docs/setup/kubernetes/upgrade/steps/)
+to do the upgrades.
+
+### Step 4. Verify the new workload certificates are loaded by Envoy
+
+You can verify your Envoy sidecars have picked on the new certificates.
+The following command can be used to check the certificates loaded for a pod _foo_ running in namespace _bar_.
```

Comment:
foo/bar may not exist.

Suggest changing to "You can verify whether an Envoy has received the new certificates. The following command shows an example to check the Envoy’s certificate for pod _foo_ running in namespace _bar_." Ideally the _valid_from_ field should be close to the current time.

Reply:
Done.